### ISSUE TITLE
when YouTubePlayerView in NestedScrollView Or RecyclerView, settings dialog can't scroll

### ISSUE THREAD

**Issue by zionFiv** [PierfrancescoSoffritti/android-youtube-player#1048]
# Bug Report

when YouTubePlayerView in NestedScrollView Or RecyclerView, settings dialog can't scroll

is there any solution ?

or what should i do if i want hide setting button

Youtube Player Library Version:  12.1.0

![EDFAA3E6-ECFB-440E-93D7-27CA05EEC3EF](https://github.com/PierfrancescoSoffritti/android-youtube-player/assets/115879302/44c6761f-5a81-423c-ae98-2effe6091b4c)

**Comment by Serkali-sudo** [comment]
I dont think there is simple solution for this problem.

But i do have some elaborate ideas: 

1) Disable recyclerview scroll when user scrolls scrollable views (only y axis is enough recyclerview doesnt intercept scrolls x axis) in youtubeplayerview.
- To do that you need to check for visibility of scrollable views in youtubeplayerview by editing html and adding callback js to java
- when you ensure that scrollable view is visible you can disable recylerview scroll and youtubeplayerview will scroll
- But after scrolling in youtubeplayerview done you have to activate recyclerview scroll again so user can scroll through videos
- To make this transcation between player and recyclerview stable and seamless you need to get coordinates(x,y) of scrollable view and touch coordinates and check if touched coordinates are in scrollable view coordinates then disable recyclerview scroll but if it is not then enable recyclerview scroll.

In summary check visibility of scrollable elements and get their coordinates check if element is visible and inside the touch coordinates then it means user wants to scroll but if the scrollable element is not visibile or not in the touched coordinates then means user wants to scroll recyclerview so enable recyclerview scroll.

Btw to implement this idea you will need to edit the html and add a callback and send touched coordinates to js ext. which means you are gonna have to edit the library

**Comment by billdizl** [comment]
> # 错误报告
> 当 YouTubePlayerView 在 NestedScrollView 或 RecyclerView 中时，设置对话框无法滚动
> 
> 有什么解决办法吗？
> 
> 或者如果我想隐藏设置按钮我该怎么办
> 
> YouTube 播放器库版本：12.1.0
> 
> ![EDFAA3E6-ECFB-440E-93D7-27CA05EEC3EF](https://private-user-images.githubusercontent.com/115879302/263211075-44c6761f-5a81-423c-ae98-2effe6091b4c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDM3MjgyOTMsIm5iZiI6MTcwMzcyNzk5MywicGF0aCI6Ii8xMTU4NzkzMDIvMjYzMjExMDc1LTQ0YzY3NjFmLTVhODEtNDIzYy1hZTk4LTJlZmZlNjA5MWI0Yy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMxMjI4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTIyOFQwMTQ2MzNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00MTk3NjcwNTJkZDI1NmU4Y2M4ZTdkMTY0MjMzZjIzYTY0MTEwYThjYmViZDUyNGE4M2M2MjgzYmEyM2EwNGIxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.57klMotBO3BT7ev-U6_8WSUuLFUXWfFIK5VD-6HGFF4)

小伙子  你可以使用自定义布局来实现隐藏

**Comment by PierfrancescoSoffritti** [comment]
Please let's stick to using english so everyone can understand.

**Comment by billdizl** [comment]
can use custom layouts to achieve hiding.
